lib/mergeDeep.js

```diff
@@ -0,0 +1,26 @@
+var isA = require('./isA');
+var cloneDeep = require('./cloneDeep');
+
+module.exports = function mergeDeep(from, to) {
+    var key;
+    for (key in from) {
+        if (from.hasOwnProperty(key)) {
+            // Property in destination object set; update its value.
+            if (isA(from[key], Object)) {
+                var mergeToObj;
+                if (to.hasOwnProperty(key) && isA(to[key], Object)) {
+                    mergeToObj = to[key];
+                } else {
+                    mergeToObj = {};
+                }
+                to[key] = mergeDeep(from[key], mergeToObj);
+            } else if (isA(from[key], Array)) {
+                to[key] = cloneDeep(from[key]);
```

Comment:
Arrays are the one case that were not handled correctly before: the new object would reference the old array. This ensures that arrays and their contents are cloned so that the original settings are not modified.